src/openrct2/core/DataSerialiserTraits.h

```diff
@@ -31,10 +38,37 @@ template<typename T> struct DataSerializerTraitsIntegral
         stream->Read(&temp);
         val = ByteSwapBE(temp);
     }
+    static void log(IStream* stream, T& val)
+    {
+        char temp[32] = {};
+        if constexpr (sizeof(T) == 1)
+            snprintf(temp, sizeof(temp), "%02X", val);
+        else if constexpr (sizeof(T) == 2)
+            snprintf(temp, sizeof(temp), "%04X", val);
+        else if constexpr (sizeof(T) == 4)
+            snprintf(temp, sizeof(temp), "%08X", val);
```

Comment:
This could use a case to handle remaining values, possibly a `static_assert` if they're not meant to be logged.